tools/python/boutiques/publisher.py

```diff
@@ -215,9 +212,105 @@ def publish(self):
             if ret.upper() != "Y":
                 return
         self.zenodo_test_api()
+
+        # perform a search to check if descriptor is an updated version
+        # of an existing one
+        from boutiques.searcher import Searcher
+        searcher = Searcher(self.descriptor.get("name"), self.verbose,
+                            self.sandbox)
+        r = searcher.zenodo_search()
+
+        id_to_update = 0
+        for hit in r.json()["hits"]["hits"]:
+            title = hit["metadata"]["title"]
+            if title == self.descriptor.get("name"):
+                id_to_update = hit["id"]
+                break
+
+        if id_to_update:
+            if(not self.no_int):
+                prompt = ("Found an existing record with the same name, "
+                          "would you like to update it? "
+                          "(Y:update existing/n:publish new version)")
+                try:
+                    ret = raw_input(prompt)  # Python 2
+                except NameError:
+                    ret = input(prompt)  # Python 3
+                if ret.upper() != "Y":
+                    self.publish_new_entry()
+            self.publish_updated_version(id_to_update)
```

Comment:
I think you're missing an `else` statement here: I tried to "publish a new version" and it created a new entry and a new version.